### ISSUE TITLE
a11y rule needed regarding order of form elements

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly#1070]
During an a11y review meeting, we discussed the need to have a rule regarding order of form elements.

For screen reader users, having ALL elements precede the submit button is important so that they realize they are there.

However, it isn't clear yet if this rule is about just the DOM order, or about visual presentation. Feedback from designers is needed before a rule can be finalized. 

Some feedback captured during this discussion:
- visually, the submit button should be last
- it would be good to be consistent with our desktop standards (which are similar to mac)
- forms in general should have width constraints ("max-width", in some cases) — sometimes that means a form that's no wider than 50rem (for example), but the whole form might be centered within a space or even left aligned, depending on context
- (in other words, I do agree that a floating button completely disconnected can be an issue, but having really wide forms is problematic in other ways as well)

**Comment by jgiardino** [comment]
The solution for #1069 might help inform the rule we define for this issue.

**Comment by jessiehuff** [comment]
It actually looks like we defined this rule in our usage and behavior guide under [forms](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/forms), section header "Button placement. Feel free to reopen if there's an additional need! 🙂
